showcase/app/controllers/components/copy/button.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { scheduleOnce } from '@ember/runloop';
+
+import type Owner from '@ember/owner';
+import type { Registry as Services } from '@ember/service';
+
+import type { ComponentsCopyButtonModelSignature } from 'showcase/routes/components/copy/button';
+
+export default class CopyButtonController extends Controller {
+  declare model: ComponentsCopyButtonModelSignature;
+  @service router!: Services['router'];
+  @tracked isModalActive = false;
+
+  constructor(owner: Owner) {
+    super(owner);
+    this.router.on('routeDidChange', this.routeDidChange.bind(this));
+  }
+
+  routeDidChange() {
+    if (this.router.currentRoute?.name === 'components.copy.button') {
+      // eslint-disable-next-line ember/no-runloop
+      scheduleOnce('afterRender', this, this.replaceMockCopyStatus.bind(this));
+    }
+  }
+
+  replaceMockCopyStatus() {
```

Comment:
Note: I moved this function into the controller because it needed to have access to `this.model`. this was working before, but when converted to TS it was undefined.